src/api/v2/v2AzureResourcesApi.ts

```diff
@@ -50,8 +50,8 @@ export interface ApplicationResourceProvider extends ResourceProviderBase<Applic
 }
 
 export interface ResourceQuickPickOptions {
-    readonly contexts?: string[];
-    readonly isParent?: boolean;
+    readonly contexts: string[];
+    readonly isLeaf: boolean;
```

Comment:
`hasChildren`?  VS Code tends to prefer parent/child terminology within the tree; while we're intending to use "branch", we may want to make sure we don't take the metaphor too far.

Reply:
I want it to be a fixed property that does not depend on whether or not the node actually has children, otherwise, we have to start querying deeper into the tree than we ought. I worried that "isParent" had the same ambiguity. However, "isLeaf" is pretty clear IMO.

We could also do something like "canHaveChildren".